src/libraries/Common/src/Interop/SunOS/procfs/Interop.ProcFs.Definitions.cs

```diff
@@ -0,0 +1,141 @@
+            public      timestruc_t pr_time;    /* usr+sys cpu time for this process */
+            public      timestruc_t pr_ctime;   /* usr+sys cpu time for reaped children */
+            public      fixed byte pr_fname[PRFNSZ];    /* name of execed file */
+            public      fixed byte pr_psargs[PRARGSZ];  /* initial characters of arg list */
+            public      int     pr_wstat;       /* if zombie, the wait() status */
+            public      int     pr_argc;        /* initial argument count */
+            private     long pr_argv;   /* address of initial argument vector */
+            private     long pr_envp;   /* address of initial environment vector */
+            private     byte    pr_dmodel;      /* data model of the process */
+            private     fixed byte pr_pad2[3];
+            public      int     pr_taskid;      /* task id */
+            public      int     pr_projid;      /* project id */
+            public      int     pr_nzomb;       /* number of zombie lwps in the process */
+            public      int     pr_poolid;      /* pool id */
+            public      int     pr_zoneid;      /* zone id */
+            public      int     pr_contract;    /* process contract */
+            private     fixed int pr_filler[1]; /* reserved for future use */
+            public      lwpsinfo pr_lwp;        /* information for representative lwp */
+        }
+
+        // Ouput type for GetThreadInfoById()
+        internal struct ThreadInfo
+        {
+            internal uint Tid;
+            internal int Priority;
+            internal int NiceVal;
+            internal char Status;
+            internal Interop.Sys.TimeSpec StartTime;
+            internal Interop.Sys.TimeSpec CpuTotalTime; // user+sys
+            // add more fields when needed.
+        }
+
+        // Ouput type for GetProcessInfoById()
+        internal struct ProcessInfo
+        {
+            internal int Pid;
+            internal int ParentPid;
+            internal int SessionId;
+            internal nuint VirtualSize;
+            internal nuint ResidentSetSize;
+            internal Interop.Sys.TimeSpec StartTime;
+            internal Interop.Sys.TimeSpec CpuTotalTime; // user+sys
+            internal string? Args;
+            // add more fields when needed.
+            internal ThreadInfo Lwp1;
+        }
+
+        internal static string GetInfoFilePathForProcess(int pid) =>
+            string.Create(null, stackalloc char[256], $"{RootPath}{(uint)pid}{psinfoFileName}");
+
+        internal static string GetLwpDirForProcess(int pid) =>
+            string.Create(null, stackalloc char[256], $"{RootPath}{(uint)pid}{lwpDirName}");
+
+        internal static string GetInfoFilePathForThread(int pid, int tid) =>
+            string.Create(null, stackalloc char[256],
+                          $"{RootPath}{(uint)pid}{lwpDirName}/{(uint)tid}{lwpsinfoFileName}");
```

Comment:
string.Create is an overhead. Simple `=> $"...";` has [better code](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHjyADAwBCMDBhhQAFPyEAlCBAwAFbBgAWaBgEsAdhgYAHbQBMtAZQxQ9AcyO49AMwgAxbQBsYAOWz4YASgYAXgA+Lh4IhgASACJ2BSVVDQBfdikAVz0Mf2MTVMMHXWc3Tx8/ZJiAbhpw7nomQQYAdWhcGBlGhJU1TR19I1MLK1t7J1cPb18A4LDqSJ5ZADoWWDV23XT3dy1cDGwwAGtsLYgwBjB1bCgAbVIAViQAXS1Y+MVulLTM/RzTfMLihMyjAKv5qtRkkA===).